### ISSUE TITLE
Training not starting

### ISSUE THREAD

**Issue by maepopi** [RVC-Project/Retrieval-based-Voice-Conversion-WebUI#2045]
Hello!

First of all, thank you for your work on that fantastic tool. 

I've got a problem on Windows though : sometimes, the training just doesn't start. I've tried twice with a dataset of 11 hours (which represents approximately 8600 clips), after preprocessing and feature extracting, nothing happens. In the console, there's no error message, there's just : 

`load model(s) from assets/hubert/hubert_base.pt

move model to cuda

all-feature-16811

all-feature-done
`

And then nothing happens. I've been leaving it for an hour but nothing moves. So I've started with a smaller dataset of about 1 hour audio, and there the training started.

I've seen in the readme that a dataset between 10mn and 50mn was recommended, so maybe I went a bit over the top there, but I saw another [thread here](https://github.com/RVC-Project/Retrieval-based-Voice-Conversion-WebUI/issues/2025) with someone who said they had trained over 4 hours audio. Have you guys tried training on really big datasets as well? Do you have an idea why the training doesn't start ? Should I try without the GUI?

Thank you very much!

**Comment by HotChocut** [comment]
I noticed that in order to train larger datasets you need to have your audio split into several smaller files, try separating your data into smaller chunks and see if that works!
_(I forgot the maximum file size for each file and it may be a different number for you, but looking at the aforementioned 4hr model the largest file I had was ~240mb in .flac format rather than .wav so I wouldn't have to split it as much)_

**Comment by maepopi** [comment]
Hey thank you @HotChocut :) Indeed I was referring to you! 

All my audios are split so that they are between 0.6 and 11 seconds. Do you think 11 seconds is still too long? Also the max size is 2Mb, and they're all wav.

If I have to resplit them, I suppose I have to make sure the sentences are not cut abruptly right? I've already split everything by silence until now.

**Comment by HotChocut** [comment]
Strange... I haven't tried training with 11hrs worth of data so maybe there's something else at work going on but the individual file sizes shouldn't be an issue with your data if it's split like that. How much storage does all your audio use total? Maybe if you convert it all into .flac format it could work. _(The the 4hr model used ~2.5gb total or something if I'm not mistaken)_

**Comment by HotChocut** [comment]
If you want you can also try and find out what the upper limit of files you can train with is, then convert the files into .flac and try again with the same storage amount but in .flac format rather than .wav. It should fit more audio than .wav without any loss in audio quality if converted properly.

**Comment by HotChocut** [comment]
Yeah, in theory you shouldn't have to train as many epochs when training with more data (especially 11hrs) so doing what I did is indeed overkill and not really recommended for most use cases. (such as this one since it would take actual days to train) I just do it so I can leave it to train overnight and not have to worry about it, though again it's not something I can recommend to anyone else.
I've never really had any issues with overtraining however despite the fact that I probably should be running into that issue with 1000 epochs. _(All of my large models were done on 1000 epochs and it's been working out great, just takes **FOREVER** to train)_

As for why they recommend less epochs on bad data sets, I'm assuming the reason is so that the model doesn't pick up on all the bad nuances of the data and train on those. Good data sets don't have that issue so it's okay to train them for more epochs.

**Comment by maepopi** [comment]
Oh Ok that makes sense! Pity there's no paper on this project, I would really love to know how it works at its core. Anyway thank you for your kind and patient answers, I'll try the reducing weight thing and let you know! I'll leave this open for now

**Comment by maepopi** [comment]
Hey there again! You were right, it was indeed a question of size. I tried launching on a dataset of 2.5Gb but it still didn't work, so I converted everything into FLAC and now the training launched! I converted with ffmpeg with the default parameters, I hope it won't affect the quality too much.

I'll try now with my big ass dataset and see if it works ^^'

**Comment by maepopi** [comment]
Okay I think the UI crashed at one point, the processing stopped and the UI wouldn't launch anymore. It might still be too heavy, and I saw a Memory error popup somewhere within the command line.

I'll train my dataset of 5000 instead of 8000, I guess it'll be fine :)

Thank you again friend,

**Comment by HotChocut** [comment]
Hopefully it works out for you, best of luck with your model!

**Comment by maepopi** [comment]
As a matter of fact, I've already had great results with as few as 200 clips and about 5 epochs, so at this stage I'm simply trying to see whether I can get even better :)

**Comment by maepopi** [comment]
Hey there! Well not really because I haven't tried training such a big dataset again. However I've been playing around with new datasets lately (small ones) and I did indeed have some trouble sometimes with the training not starting. I think it was mostly because of the "one click training" button. Try and do all the steps manually (preprocess, feature extraction, training, then train index), and don't touch the one click training button.

Let me know how it goes !

**Comment by 240db** [comment]
So this project cannot train big datasets? That is a big warning to have in the docs though, I am facing several issues like described in this thread as I am trying to recycle data I used to train other TTS models such as xtts. To be specific, using the model didnt work because it was missing the .index file. I even tried looking up for it to see if maybe i could create one for xtts model, but didnt find anything. 

Instead, tried to train my own model but like @maepopi  said, "one click training" basically does nothing. Clicking training also issues an error on Gradio but shows terminal running epochs on terminal, except way too fast ? Then trying Train Feature Index just does not do anything at all.

Im gonna try running it again with https://github.com/daswer123/xtts-finetune-webui to create my datasets, as it splits them into smaller chunks. Then i will try to make that training set even smaller, mine was around 220mb as well but each audio was like 18mb with at least 2 minutes (that is what xtts-finetune-webui requires)

**Comment by 240db** [comment]
The gradio app seems to be buggy. Also, I managed to get it work once, but when completed the inference, it said it didnt use the index file. After that, tried again and nothing worked. Will only work if ''harvest'' is chosen during training and inference..

**Comment by Ultramarine1939-syujie** [comment]
Fix: Add safe globals for fairseq Dictionary to prevent model loading error

After add these code, this problem shall not happen again

```python
import torch
try:
    from fairseq.data.dictionary import Dictionary
    if hasattr(torch.serialization, 'add_safe_globals'):
        torch.serialization.add_safe_globals([Dictionary])
except:
    pass
